### PR TITLE
Add .editorconfig

### PR DESCRIPTION
**Pull request by nasivuela**
A basic `.editorconfig` to maintain consistent coding styles.
[Docs](http://editorconfig.org/)

Suggestions are welcome!

### CODE REVIEW

File: .editorconfig [hunk 1]
```diff
@@ -0,0 +1,21 @@
+# EditorConfig is awesome: http://EditorConfig.org
+
+# top-most EditorConfig file
+root = true
+# Unix-style newlines with a newline ending every file
+[*]
+end_of_line = lf
+insert_final_newline = true
+
+# 4 space indentation
```
[the remark this answers is not in the document]
Comment (nasivuela) [reply .editorconfig]: Cierto!

**Comment by nasivuela** [comment]
Agreed!

**Comment by guerrero** [comment]
Nice idea to add `.editorconfig` to this repo 👍 . I want to leave here a couple of comments to discuss:

IMHO It would be nice to add the following role in order to preserve the option to create `<br>` tags in Markdown syntax

```
[*.md]
trim_trailing_whitespace = false
```
In Markdown, you can place two spaces at the end of a line to create a `<br>`. `hello  __\nworld` would be transformed to `hello<br>world` but if the trailing spaces are removed the output wold be `hello\nworld`.

Another small thing is that it is not necessary to add multiple `[*]`. Regarding your file there's only one needed to apply all the rules and the code would be like the following:

```
# EditorConfig is awesome: http://EditorConfig.org


# top-most EditorConfig file
root = true


[*]

# Unix-style newlines with a newline ending every file
end_of_line = lf
insert_final_newline = true

# 2 space indentation
indent_style = space
indent_size = 2

# utf-8 character set
charset = utf-8

# Remove any whitespace characters preceding newline characters
trim_trailing_whitespace = true


[*.md]

# Preserve trailing whitespace on Markdown to preserve <br>s
trim_trailing_whitespace = false
```

**Comment by nasivuela** [comment]
@guerrero Good catch!
I added [*] to make clear that it can be customized, but having a real one is much better!
Will update it soon.

**Pull request merged**